site/content/blog/2025-02-17-portswigger-lab-walkthrough-broken-brute-force-protection-ip-block/index.md

```diff
@@ -0,0 +1,793 @@
+---
+title: "PortSwigger Lab Walkthrough: Broken Brute-Force Protection, IP Block"
+summary: >
+  Walkthrough for the PortSwigger lab, "Broken brute-force protection, IP block".
+# images:
+# - 
+type: post
+tags:
+- blog
+- fuzzer
+- portswigger-lab
+- javascript
+date: "2025-02-17"
+authors: 
+    - chibbyalucard
+---
+
+This article describes how to solve the PortSwigger lab, “[Broken brute-force protection, IP block](https://portswigger.net/web-security/authentication/password-based/lab-broken-bruteforce-protection-ip-block)”, using ZAP rather than Burp Suite. You can find this lab in the [Authentication vulnerabilities](https://portswigger.net/web-security/authentication) learning path at the PortSwigger Web Security Academy.
+
+If you’re familiar with Burp Suite already, it might be helpful to take a look at the [Burp to ZAP Feature Map](https://www.zaproxy.org/docs/burp-to-zap-feature-map/) for a mapping of Burp features to ZAP equivalents. However, this article assumes little to no familiarity with ZAP, and I’ve attempted to make it as easy as possible to follow along.
+
+## Prerequisites
+
+To continue with the rest of the article, you’ll need the following:
+
+1. A [PortSwigger Academy](https://portswigger.net/web-security) account
+2. A [ZAP](https://www.zaproxy.org/download/) installation
+3. Basic JavaScript knowledge (See [Scripts](https://www.zaproxy.org/docs/desktop/start/features/scripts/) for other scripting languages you can use in ZAP)
+
+If you’re all set, let’s jump in!
+
+## Have Username, Want Password
+
+We’ll begin with a review of the lab’s instructions:
+
+> This lab is vulnerable due to a logic flaw in its password brute-force protection. To solve the lab, brute-force the victim's password, then log in and access their account page.
+> 
+> - Your credentials: `wiener:peter`
+> - Victim's username: `carlos`
+> - [Candidate passwords](https://portswigger.net/web-security/authentication/auth-lab-passwords)
+
+Brute-force protection is especially important for applications that depend entirely on password-based authentication. Without additional guard rails like multi-factor authentication (MFA), an attacker who can make multiple or unlimited guesses of users’ login credentials could eventually gain access to their accounts.
+
+However, poor implementation can undermine even the best defense mechanisms. If developers make incorrect assumptions about how users will interact with their application, they may unintentionally introduce vulnerabilities that attackers can exploit.
+
+This can be in the form of a logic flaw — a vulnerability that allows users to manipulate normal application functionality or workflow to achieve results the developer neither intended nor expected.
+
+In this lab, we’ll exploit a logic flaw to access another user’s account. The lab provides us with a specific username, “carlos”. So, we don’t need to guess a valid username — we can focus on brute-forcing the password.
```

Comment:
"...brute-forcing carlos' password."

Reply:
I've suggested "carlos'" which you seem to have used quite a few places as well. However there's also some "carlos's". I'm not terribly set either way, but I'd suggest being consistent so do a search and replace one way or the other :wink: